### PR TITLE
[msbuild] Add desugar support.

### PR DESCRIPTION
**Pull request by atsushieno**
This should fix bug #60336.

This is only for release builds.

### CODE REVIEW

**Comment by atsushieno** [comment]
Marking as DO-NOT-MERGE because repro at bug #60336 is incomplete and there is no way to verify further.

File: src/Xamarin.Android.Build.Tasks/Tasks/Desugar.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+				cmd.AppendSwitch ("--input ");
+				cmd.AppendFileNameIfNotNull (jar);
+				cmd.AppendSwitch ("--output ");
+				cmd.AppendFileNameIfNotNull (output);
+			}
+
+			OutputJars = outputs.ToArray ();
+			Log.LogDebugTaskItems ("  OutputJars: ", OutputJars);
+
+			return cmd.ToString ();
```
Comment (DeKaN) [inline src/Xamarin.Android.Build.Tasks/Tasks/Desugar.cs]: You've forgot to add parameter `--min_sdk_version`. It is important for desugar to know application minimum sdk version (default is 1).

Comment (atsushieno) [reply src/Xamarin.Android.Build.Tasks/Tasks/Desugar.cs]: Weird, we have a test that does pass without it. Could you elaborate and explain why/how it is important?

Comment (DeKaN) [reply src/Xamarin.Android.Build.Tasks/Tasks/Desugar.cs]: > There are multiple reasons but the main one is that we need to know the application's minSdk to desugar correctly. Also, there is potentially a classpath issue when dealing with composite builds and multiple app importing the same library module. So we do all the desugaring in the main app project once we know the above information.

https://issuetracker.google.com/issues/67452341#comment2

For example, if minSdk < 19, then desugar replace `try-with-resources` to simple `try..catch`

Comment (atsushieno) [reply src/Xamarin.Android.Build.Tasks/Tasks/Desugar.cs]: Oh, that sounds correct. Okay, we should work on that. Thanks for the info!

**Pull request merged**